packages/kbn-unified-data-table/src/components/data_table.tsx

```diff
@@ -375,6 +375,9 @@ export interface UnifiedDataTableProps {
    * Optional extra props passed to the renderCellValue function/component.
    */
   cellContext?: EuiDataGridProps['cellContext'];
+
+  /* Optional props passed to Columns to display Provided Labels as Column name instead of field name */
+  columnHeaders?: Record<string, string>;
```

Comment:
I think we can improve this description to include the information on what to pass, and that these labels will take precedence over the data view field names.

/**
 * Optional props passed to Columns to display provided labels as column names instead of field names.
 * This object maps column field names to their corresponding display labels.
 * These labels will take precedence over the data view field names.
 */
columnHeaders?: Record<string, string>;